python/pyspark/sql/context.py

```diff
@@ -204,15 +206,31 @@ def registerFunction(self, name, f, returnType=StringType()):
         >>> sqlContext.sql("SELECT stringLengthInt('test')").collect()
         [Row(stringLengthInt(test)=4)]
 
+        >>> from pyspark.sql.types import IntegerType
+        >>> from pyspark.sql.functions import udf
+        >>> slen = udf(lambda s: len(s), IntegerType())
+        >>> _ = sqlContext.udf.register("slen", slen)
+        >>> sqlContext.sql("SELECT slen('test')").collect()
+        [Row(slen(test)=4)]
+
         >>> import random
         >>> from pyspark.sql.functions import udf
-        >>> from pyspark.sql.types import IntegerType, StringType
+        >>> from pyspark.sql.types import IntegerType
         >>> random_udf = udf(lambda: random.randint(0, 100), IntegerType()).asNondeterministic()
-        >>> newRandom_udf = sqlContext.registerFunction("random_udf", random_udf, StringType())
+        >>> newRandom_udf = sqlContext.udf.register("random_udf", random_udf)
```

Comment:
Would it be better to keep `sqlContext.registerFunction` as was? The documentation will show the examples for `SQLContext.registerFunction` API ..

Reply:
Maybe you can submit a separate PR for it? For testing purpose, we should do it in a test suite instead of using doc.

Reply:
Doctest is for testing purpose too. I intended to do this in a separate PR and that's why I suggest to leave it as was.

Reply:
The current change keeps `sqlContext`. We just need to add a test case for `sqlContext.registerFunction`

Reply:
I mean it doesn't completely cover the concern:

> `sqlContext` has been deprecated since 2.0. SparkSession should be the default entrance

and this change doesn't completely replace it too. If it's meant to be separate, we should better leave this change out. What I was wondering is why this partially fixes this concern in a separate PR.

Reply:
I will do it in this PR.

Reply:
I beg to just leave it as was ..

Reply:
I can revert it back if you want to take it.

Reply:
Sure, I want to take it. Thanks.